hathor/consensus/consensus.py

```diff
@@ -111,7 +113,8 @@ def _unsafe_update(self, base: BaseTransaction) -> None:
             if to_remove:
                 self.log.warn('some transactions on the mempool became invalid and will be removed',
                               count=len(to_remove))
-                storage.remove_transactions(to_remove)
+                with storage.allow_invalid_context():
```

Comment:
Add a note explaining that the mempool cleaning mechanism invalidates all transactions so we need to allow invalid transactions here.

Reply:
Great! Thanks!